### ISSUE TITLE
parsed only first object from JS file

### ISSUE THREAD

**Issue by Roman-Kasianenko** [Nykakin/chompjs#42]
if we have JS like this:
```
const mapData = {
  projection: 'E',
  latestWkid: 2039,
  centerX: 241610,
  centerY: 664717,
  extent: true, 
  minx: 60147,
  miny: 516486,
  maxx: 551144,
  maxy: 938226,
  min_zoom: 10.5, 
  max_zoom: 23,
  default_zoom: 11
}

const settings = {
  defaultLanguage: 'HE',
  isMapComparison: true,
  email: '',
  login: false,
  basepath: '/public',
  LayersOpacity: 0.75,               
  defaultBuffer: 50,
  maxBuffer: 2000,
  stepBuffer: 5,
  minBuffer: 0,
  isZoomGeolocation: true,
  zoomGeolocation: 18,
}

const languageMenu = {
  HE: true,
  EN: true
}

export {
  mapData,
  settings,
  languageMenu
}

```

after parsing in JSON we will have only this one:

```
{
  "projection": "E,
  "latestWkid": 2039,
  "centerX": 241610,
  "centerY": 664717,
  "extent": true,
  "minx": 60147,
  "miny": 516486,
  "maxx": 551144,
  "maxy": 938226,
  "min_zoom": 10.5,
  "max_zoom": 23,
  "default_zoom": 11
}
```

as you can see, `settings` and `languageMenu ` not present in json.
could you please let me know if you can fix it or if there is any workaround to avoid this issue?

**Comment by Nykakin** [comment]
This is the expected behavior: only the first JSON-like thing in the input script is parsed. If you have more, then youneed to first extract the relevant fragment yourself, usually with a regex. In your case you can do:
```python
import chompjs
import re

data = """
const mapData = {
  projection: 'E',
  latestWkid: 2039,
  centerX: 241610,
  centerY: 664717,
  extent: true, 
  minx: 60147,
  miny: 516486,
  maxx: 551144,
  maxy: 938226,
  min_zoom: 10.5, 
  max_zoom: 23,
  default_zoom: 11
}

const settings = {
  defaultLanguage: 'HE',
  isMapComparison: true,
  email: '',
  login: false,
  basepath: '/public',
  LayersOpacity: 0.75,               
  defaultBuffer: 50,
  maxBuffer: 2000,
  stepBuffer: 5,
  minBuffer: 0,
  isZoomGeolocation: true,
  zoomGeolocation: 18,
}

const languageMenu = {
  HE: true,
  EN: true
}

export {
  mapData,
  settings,
  languageMenu
}
"""

result = []
for fragment in re.findall(r"{.*?}", data, re.DOTALL):
    try:
        result.append(chompjs.parse_js_object(fragment))
    except ValueError:
        pass

print(result)
```
At makes sense to attempt retrieving all potential JSONs so I've opened a separate issue: https://github.com/Nykakin/chompjs/issues/43 to add a second extracting function with this functionality.

**Comment by Nykakin** [comment]
Closed in https://github.com/Nykakin/chompjs/pull/45

```python
>>> data = '''const mapData = {
...   projection: 'E',
...   latestWkid: 2039,
...   centerX: 241610,
...   centerY: 664717,
...   extent: true, 
...   minx: 60147,
...   miny: 516486,
...   maxx: 551144,
...   maxy: 938226,
...   min_zoom: 10.5, 
...   max_zoom: 23,
...   default_zoom: 11
... }
... 
... const settings = {
...   defaultLanguage: 'HE',
...   isMapComparison: true,
...   email: '',
...   login: false,
...   basepath: '/public',
...   LayersOpacity: 0.75,               
...   defaultBuffer: 50,
...   maxBuffer: 2000,
...   stepBuffer: 5,
...   minBuffer: 0,
...   isZoomGeolocation: true,
...   zoomGeolocation: 18,
... }
... 
... const languageMenu = {
...   HE: true,
...   EN: true
... }
... 
... export {
...   mapData,
...   settings,
...   languageMenu
... }
... '''
>>> 
>>> import chompjs
>>> it = chompjs.parse_js_objects(data)
>>> next(it)
{'projection': 'E', 'latestWkid': 2039, 'centerX': 241610, 'centerY': 664717, 'extent': True, 'minx': 60147, 'miny': 516486, 'maxx': 551144, 'maxy': 938226, 'min_zoom': 10.5, 'max_zoom': 23, 'default_zoom': 11}
>>> next(it)
{'defaultLanguage': 'HE', 'isMapComparison': True, 'email': '', 'login': False, 'basepath': '/public', 'LayersOpacity': 0.75, 'defaultBuffer': 50, 'maxBuffer': 2000, 'stepBuffer': 5, 'minBuffer': 0, 'isZoomGeolocation': True, 'zoomGeolocation': 18}
>>> next(it)
{'HE': True, 'EN': True}
>>> next(it)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
>>> 
```
Currently available on master, I should release version 1.2.0 in PyPI soon.

**Comment by Nykakin** [comment]
Closing as version 1.2.0 got released. Thank you for the issue!
